README.md

```diff
@@ -182,6 +196,43 @@ This is useful if you have control over secret creation and would like to avoid
 
 Example: `buildkite-secret`
 
+### `external-secrets` (optional, string or array)
+
+Mount one or many secrets as a directory inside the container from an external source using [ExternalSecrets](https://external-secrets.io/v0.7.2/). Must be in the form `secretKey:externalProperty:externalKey`.
+
+`external-secrets-mount-path` can be set to change where the secrets are mounted, but they must all be mounted to the same location.
+`secret-store` or `cluster-store` must be defined to specify what existing SecretStore to pull data from. Only one can be defined.
+
+Example
+```yaml
+secret-store: vault-backend
+external-secrets:
+- "ciPassword:secret/dev/ci:ci-password"
+- "ciUser:secret/dev/ci:ci-user"
+```
+
+Will create an ExternalSecret resource pulling from the vault-backend Secret Store, which will create a secret with keys `ciPassword` and `ciUser` using the values from the `secret/dev/ci` properties and `ci-password` and `ci-user` keys.
+
+**This has only been tested with Vault, other keystores may or may not function as expected**
+
+### `secret-store` (optional, string)
+
+Describes the SecretStore to pull ExternalSecrets from. 
+
+Must be used with the ``external-secrets`` argument and cannot be used with the `cluster-store` argument
+
+### `cluster-store` (optional, string)
+
+Describes the ClusterStore to pull ExternalSecrets from. 
+
+Must be used with the ``external-secrets`` argument and cannot be used with the ``secret-store`` argument
+
+### `mount-path-external-secrets` (optional,string)
+
+Sets the mount path of the external secrets 
+
+Default: ``/externalsecrets``
```

Comment:
I don't know enough about the controller, but is it likely to have both cluster-stores and secret-stores? If so; having them be mutually exclusive across a workflow seems awkward. It also - maybe - seems like across a larger org you might have multiple sources of secrets? I'm not sure what the BK plugin config supports; but maybe we could have some form of hierarchical setup?

```yaml
external-secrets:
- store: some-backend
  kind: ClusterStore
  secrets:
  - name: ciPassword
    secretName: secret/dev/ci
    key: ci-password
```
 
I'd be in favor of a less smart, more explicit configuration here.

Reply:
It is likely to have both secret-stores and cluster-stores in the same cluster, but you can only reference one store per ExternalSecret.  A "limitation" of this implementation is that everything is put into one ExternalSecret resource, so only a ClusterStore or a SecretStore can be referenced.

I have not found the ability to do hierarchal configuration in buildkite, which has been a major pain point and I really wish I could.

Reply:
Gotcha. I guess one could do something like
```yaml
external-secrets: |
  yaml-as-string
```
but it'd likely be a lot more fragile. Gotcha on the limitations; I'll have a think.

Reply:
I thought this as well, but I'm reluctant to do yaml parsing in the plugin, I think it does make the plugin a lot more fragile, but honestly I think the system that Buildkite presents is fragile as well, and I don't think there's any way to get around that without a major rethinking of how to pass configuration.